ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -171,7 +171,7 @@ bracketChainSyncClient tracer ChainDbView { getIsInvalidBlock } varCandidates
 -- We must keep the candidate chain synchronised with the corresponding
 -- upstream chain. The upstream node's chain might roll forward or
 -- backwards, and they will inform us about this. When we get these
--- messages, we will replicate these actions on our candidate chain.
+-- messages, we will replicate these actions on our fragment.
```

Comment:
Adopted in 93395a3ce.

Reply:
I'm sorry this didn't occur to me earlier, but: there are plenty of other occurrences of "candidate chain" in the rest of the comments in this source file.

Ultimately this is because engineers on this project eventually realize that "fragment" and "chain" are essentially interchangeable. It seems that whoever originally wrote this module was favoring "chain" at the time. Some other comments do already say "candidate fragment". I would guess that---if anything---it's most correlated to how broad the context of the comment is: "chains" in the broad cases, "fragment" in the very specific cases.

Usually, I wouldn't object to your PR as-is now: you've ensured the wording is at least coherent within its local context.

However, in this case, that local context happens to effectively be the binding site of "candidate chain" as it occurs throughout the rest of the module.

However however, the module was already using both "candidate chains" and "candidate fragment"... and it doesn't really matter which is used where. The only possible downside to your PR is that it inverts my hypothesized correlation between scope and chain-versus-fragment word choice 🤷.

Reply:
Thanks for paying attention @nfrisby; in my initial review, I somehow thought that the context was the same for both changes in https://github.com/input-output-hk/ouroboros-consensus/pull/233/commits/982fc1c3702d2e47d5bb0b2b2c56187dd4ebb76b. I agree that the choice of "candidate chain" vs "candidate fragment" doesn't matter that much.

I am both fine with reverting the change in this block comment, or keeping it, in that case probably adding "candidate" again
```suggestion
-- messages, we will replicate these actions on the candidate fragment.
```
to make it clear that this is in fact *not* referring to the `ourFrag` variable.

Reply:
@facundominguez [wrote](https://github.com/input-output-hk/ouroboros-consensus/pull/233#issuecomment-1642100343):

> I'm finding useful to talk about "candidate fragment" when referring to the local anchored fragment, and "candidate chain" to refer to the "upstream chain". My last commit https://github.com/input-output-hk/ouroboros-consensus/commit/26c999ac0810dd0dca1734db469e161d66271175 makes this story consistent in Client.hs.
> 
> To make this even more obvious, we could rename "candidate chain" to "upstream chain" if you want.
> 
> Update: I think considering "candidate fragment" and "candidate chain" interchangeable would really hide whether the text is referring to the upstream chain or to "their fragment" which is locally kept.

I haven't fully processed this. But my first knee-jerk reaction is that it seems while to point out: from a practical perspective, the upstream chain is almost always irrelevant to the code, simply because we have no way of observing it. The only thing we can observe is the candidate fragment, ie the prefix of the upstream chain that they have sent so far. The only knowledge we have about the upstream chain is that it is some (perhaps non-proper) extension of the candidate fragment. (Exception: they also send the tip point annotation and `MsgAwaitReply`, but those merely determine whether it's a proper or non-proper extension.)

Reply:
> the upstream chain is almost always irrelevant to the code, simply because we have no way of observing it.

In that case, I invite you to check the remaining references to "candidate chain". I think they all refer to the upstream chain despite of lacking a way to observe it.